internal/mesh/internal/controllers/explicitdestinations/controller.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package explicitdestinations
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/mappers/workloadselectionmapper"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v2beta1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const ControllerName = "consul.io/explicit-destinations-controller"
```

Comment:
If this is supposed to merge explicit/implicit destinations, why name it for only the explicit side?

Reply:
I meant this only for explicit destinations. There should be a separate controller that computes implicit destinations from intentions and stores it in a different resource. Maybe I should rename the computed resource to `ComputedExplicitDestinations`?

Reply:
I think if you intended for them to be different then yes that makes sense.

I guess the alternative would be to have `ComputedDestinations` have two top level slice fields `Explicit` and `Implicit` and have this one controller create both sides at the same time.

Reply:
Yeah we could do that but I worry that this controller will then become more complex because originally I intended implicit destinations resource to be workload-identity aligned rather than workload name aligned. At least for now, I think I'd lean to keep them separate.